indico/modules/designer/controllers.py

```diff
@@ -268,14 +273,33 @@ def _process(self):
             return jsonify(error="File format not accepted!")
         content_type = 'image/' + image_type
         image = DesignerImageFile(template=self.template, filename=filename, content_type=content_type)
-        self.template.background_image = image
         image.save(data)
         flash(_("The image has been uploaded"), 'success')
-        return jsonify_data(image_url=image.download_url)
+        return jsonify_data(image_url=image.download_url, image_id=image.id)
 
 
 class RHDeleteDesignerTemplate(RHModifyDesignerTemplateBase):
     def _process(self):
         db.session.delete(self.template)
         flash(_('The template has been removed'), 'success')
         return jsonify_data(html=_render_template_list(self.target, event=self.event_or_none))
+
+
+class RHListBackSideTemplates(RHModifyDesignerTemplateBase, RHListEventTemplates):
+    def _process(self):
+        inherited_templates = [tpl for tpl in get_inherited_templates(self.target)
+                               if not tpl.backside_template and tpl.type == TemplateType.badge]
+        custom_templates = [tpl for tpl in self.target.designer_templates
+                            if not tpl.backside_template and tpl != self.template]
+        return jsonify_template('designer/backside_list.html', target=self.target, custom_templates=custom_templates,
+                                inherited_templates=inherited_templates, current_template=self.template)
+
+
+class RHGetTemplateData(RHModifyDesignerTemplateBase):
+    def _process(self):
+        template_data = {
+            'title': self.template.title,
+            'data': self.template.data,
+            'background_url': self.template.background_image.download_url if self.template.background_image else None
+        }
+        return jsonify_data(template=template_data, backside_template_id=self.template.id)
```

Comment:
`flash=False` or just use `jsonify()` since you don't really care about the `success: true` in the response